p2p/SCORING.md

```diff
@@ -0,0 +1,406 @@
+
+As mentioned above, this solution is based on ETH2 
+(see [function](https://gist.github.com/blacktemplar/5c1862cb3f0e32a1a7fb0b25e79e6e2c#file-generate-scoring-params-py-L144))
+for calculating peer score params.
+
+```json
+{
+  "TopicScoreCap": 56.25, // = maxPositiveScore / 4.0
+  "AppSpecificWeight": 1, // not used for now
+  "IPColocationFactorWeight": -56.25, // = -topicScoreCap
+  "IPColocationFactorThreshold": 10,
+  "IPColocationFactorWhitelist": [],
+  "BehaviourPenaltyWeight": -47.74333136903669, // = gossipThreshold / decayConvergence(rate = 10.0) ^ 2
+  "BehaviourPenaltyThreshold": 10,
+  "BehaviourPenaltyDecay": 0.6309573444801931,  // = decay oneEpoch*10
+  "DecayInterval": 384000000000, // = oneEpoch
+  "DecayToZero": 0.01,
+  "RetainScore": 3840000000000, // = oneEpoch * 10
+  "SeenMsgTTL": 385000000000
+}
+```
+
+#### Input Arguments
+
+The following arguments can be provided as an input to peer score params procedure:
+
+- one epoch duration should set the time offset for scoring (`12 seconds * 32 slots`)
+- msgID cache TTL
+- IP colocation threshold the amount of peers allowed from the same IP (SSV default `10`)
+- IP colocation weight helps to determine the penalty for multiple peers of the same IP
+- IP whitelist to enable specific peers
+
+<br />
+
+## Topic Score Params
+
+Topic score params enables to define some more specific values in the topic level, 
+ensuring that each topic has its own set of config according to the topic nature.
+
+As mentioned above, our solution for calculating score params is based on ETH2
+[procedure](https://gist.github.com/blacktemplar/5c1862cb3f0e32a1a7fb0b25e79e6e2c#file-generate-scoring-params-py-L174),
+with some minor adaptations to ssv.
+
+Topic score params accepts multiple types of arguments:
+
+**Network level arguments**
+
+* `ActiveValidators` is the amount of validators in the network
+* `Subnets` is the number of subnets in the network
+  * `Groups` is the amount of groups used in the network. **TBD** pending network topology 
+* `OneEpochDuration` is used as a time-frame length to control scoring in a dynamic way
+* `TotalTopicsWeight` is the weight of all the available topics in the network (decided + subnets)
+
+**Topic level arguments**
+
+* `TopicWeight` is the weight of the topic
+* `ExpectedMsgRate` is the expected rate for the topic (in 12sec): \
+`activeValidators / float64(slotsPerEpoch)`
+* `InvalidMsgDecayTime` defines the decay for invalid messages (P4),
+passing a zero value disables scoring of message validation.
+* `FirstMsgDecayTime` defines the decay time for first message deliveries (P2)
+* `MeshMsgDecayTime` defines the decay time for mesh message deliveries (P3)
+  * `MeshMsgCapFactor` defines the factor to use to apply on the mesh message deliveries cap
+  * `MeshMsgActivationTime` defines time in the mesh before penalties are being applied
+* `D` is the gossip degree of the topic
+
+### Decided Topic Params
```

Comment:
did we take the way we calculate from somewhere? or ?

Reply:
most of our stuff is based on ETH2, the topic params are based on (mentioned few lines above):
* [example params for 51k validators](https://gist.github.com/blacktemplar/5c1862cb3f0e32a1a7fb0b25e79e6e2c#file-51000-toml)
* [procedure for calculating topic params](https://gist.github.com/blacktemplar/5c1862cb3f0e32a1a7fb0b25e79e6e2c#file-generate-scoring-params-py-L174)